### ISSUE TITLE
[Truffle] Implement InvokeRubiniusPrimitiveNode annotation properties

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#2882]
Rubinius.invoke_primitive should implement lowerFixnumParameters and needsSelf.

This is not yet implemented in the BodyTranslator.

**Comment by chrisseaton** [comment]
Sorry I'm not really sure what you mean here. Can you rephrase or give an example of the problem?

**Comment by bjfish** [comment]
@chrisseaton lowerFixnumParameteres works for  `Rubinius.primitive` calls but not for `Rubinius.invoke_primitive` DOes that make sense?

**Comment by bjfish** [comment]
@chrisseaton A more concrete example:
```
./bin/jruby -X+T -e'a = [0,2**35];m = Rubinius::Mirror.reflect "test";m.copy_from self, a[0], 1, a[0]'
core:/core/rubinius/common/string_mirror.rb:53:in `copy_from': Truffle doesn't have a case for the org.jruby.truffle.nodes.rubinius.StringPrimitiveNodesFactory$StringCopyFromPrimitiveNodeFactory$StringCopyFromPrimitiveNodeGen node with values of type  String(org.jruby.truffle.runtime.core.RubyString) Object(org.jruby.truffle.runtime.core.RubyBasicObject) java.lang.Long=0 java.lang.Integer=1 java.lang.Long=0 (TypeError)
	from -e:1:in `<main>'
```
If I then add: ` lowerFixnumParameters = {2 , 3, 4}` to `StringCopyFromPrimitiveNode` it still shows the same error. It looks like the `translateRubiniusInvokePrimitive` method in `BodyTranslator` doesn't implement the annotations.

**Comment by chrisseaton** [comment]
I think this is fixed in 1a706b92a4dddcbcb90de573ed767ec947cc1254, but I still can't run your test case because you are copying from self, which isn't a string, so there still isn't a suitable specialisation.
